### ISSUE TITLE
Error reading NIC DHCP IP from NSX-V in vcd_vapp_vm  (network_dhcp_wait_seconds parameter)

### ISSUE THREAD

**Issue by rainerleber** [vmware/terraform-provider-vcd#666]
I have an error. I want to create a vm in a second vdc with a shared network, spanned over two vdcs with one attached nsx. On the nsx dhcp for the network is configured.

The provider want to read the applied dhcp ip from the nsx in vdc "a" the vm is created in vdc "b". The gateway entries in vdc "b" are empty this is ok because i want to use the nsx from vdc "a".

Please implement an other way to read the DHCP ip. 

### Terraform Version
Tested on 0.14.x and 0.15.x

### Affected Resource(s)

vcd_vapp_vm

### Terraform Configuration Files
```
doesn't matter can be a default deployment with one vapp and a vapp_vm attached to the network described below.
```

### Debug Output
╷
│ Error: [VM read] failed reading network details: unable to to lookup DHCP IPs for VM NICs '[0]': unable to validate if NICs are attached to edge gateway: could not lookup edge gateway for routed network on NIC 0: [ENF] entity not found
│
```

### Expected Behavior
Return DHCP IP

### Actual Behavior
crash on fetching DHCP because provider want to contact the nsx in an other vdc.

### Steps to Reproduce
1. Two vdcs
  - First vdc with nsx.
  - Second vdc no nsx
2. one shared network over both vdcs with attached nsx

1. `terraform apply` the vm to the second vdc

### Important Factoids
-

### References
the error maybe cloud be here:
https://github.com/vmware/terraform-provider-vcd/blob/db8d6c9a69563a881a783507037bd2b5aebdad9b/vcd/resource_vcd_vapp_vm.go#L2025

**Comment by dataclouder** [comment]
Could you please add an HCL script that reproduces the error?

**Comment by rainerleber** [comment]
```
terraform {
  required_providers {
    vcd = {
      source  = "vmware/vcd"
      version = "=3.2.0"
    }
  }
  required_version = ">= 0.13"
}

variable "vcd_user" {
  type        = string
  description = "VCD username to log in (aaa.bbb@vdc.adm)"
}

variable "vcd_password" {
  type        = string
  description = "VCD password to log in"
}

variable "vcd_org" {
  type        = string
  description = "VCD organisation name to log in"
}

variable "vcd_url" {
  type        = string
  description = "VCD URL to log in"
}

variable "vcd_dc_a" {
  type = string
}

variable "vcd_dc_b" {
  type = string
}

provider "vcd" {
  user     = var.vcd_user
  password = var.vcd_password
  org      = var.vcd_org
  url      = var.vcd_url
  vdc      = var.vcd_dc_a
  #  auth_type = "saml_adfs"
}

provider "vcd" {
  user     = var.vcd_user
  password = var.vcd_password
  org      = var.vcd_org
  url      = var.vcd_url
  vdc      = var.vcd_dc_b
  alias    = "vcd-b"
  #  auth_type = "saml_adfs"
}

resource "vcd_network_routed" "net" {
  org          = var.vcd_org
  vdc          = var.vcd_dc_a
  description  = "example"
  name         = "net-prod-001"
  edge_gateway = var.gw
  gateway      = cidrhost("10.2.2.0/24", 1)
  netmask      = cidrnetmask("10.2.2.0/24")
  dns1         = "1.1.1.1"
  dns2         = "8.8.8.8"
  dns_suffix   = "test1.org"
  shared       = true
  dynamic "dhcp_pool" {
    for_each = length(var.dhcp_range) > 0 ? [1] : []
    content {
      start_address = cidrhost("10.2.2.0/24", var.dhcp_range[0])
      end_address   = cidrhost("10.2.2.0/24", var.dhcp_range[1])
    }
  }
}

resource "vcd_vapp" "vm" {
  depends_on = [vcd_network_routed.net]
  provider   = vcd.vcd-b
  name       = "vap-vm"
}

resource "vcd_vapp_org_network" "vm" {
  depends_on       = [vcd_vapp.vm]
  provider         = vcd.vcd-b
  vapp_name        = vcd_vapp.vm.name
  org_network_name = "net-prod-001"
}

resource "vcd_vapp_vm" "vm" {
  vapp_name        = vcd_vapp.vm.name
  depends_on       = [vcd_vapp_org_network.vm]
  provider         = vcd.vcd-b
  name             = lower("vm-001")
  catalog_name     = example-catalog
  template_name    = template-gen
  memory           = 2048 * 1024
  cpus             = 2
  cpu_cores        = 1
  os_type          = "sles12_64Guest"
  hardware_version = "vmx-14"

  cpu_hot_add_enabled    = true
  memory_hot_add_enabled = true

  dynamic "network" {
    content {
      type               = "org"
      adapter_type       = "VMXNET3"
      name               = vcd_vapp_org_network.vm.org_network_name
      ip_allocation_mode = DHCP
      ip                 = ""
      is_primary         = true
    }
  }

  network_dhcp_wait_seconds = 60

  override_template_disk {
    size_in_mb  = 30 * 1024
    bus_type    = "ide"
    bus_number  = 0
    unit_number = 0
  }

  customization {
    enabled = true
  }

  metadata = var.metadata

}
```

**Comment by rainerleber** [comment]
above the requested Terraform Code

**Comment by Didainius** [comment]
Just two more question:
* Are you on the latest provider version (3.2)>
* Do you require `network_dhcp_wait_seconds`? If not - you could just not use the parameter and it should not trigger it.

**Comment by rainerleber** [comment]
Hi, 
thank you for your quick reply.
both yes 
The used provider is 3.2
I need network_dhcp_wait_seconds to output the ip.

**Comment by Didainius** [comment]
And is this NSX-V or NSX-T you have?

**Comment by rainerleber** [comment]
Yes, it is a NSX-V

**Comment by rainerleber** [comment]
Are you investigating in this issue we really need this feature:-)

**Comment by Didainius** [comment]
Hello,
I have not yet had time to focus on it, but this is not forgotten

**Comment by vbauzysvmware** [comment]
Will be released with 3.7.0 version.
